routers/organizationRouter.go

```diff
@@ -296,3 +297,54 @@ func (handler *organizationHandler) RemoveMemberFromOrganization(writer http.Res
 	writer.Header().Set("Content-Type", "application/json")
 	writer.WriteHeader(http.StatusNoContent)
 }
+
+func (handler *organizationHandler) GetAllBoards(writer http.ResponseWriter, request *http.Request) {
+	params := mux.Vars(request)
+	organizationId := params["organizationId"]
+	if organizationId == "" {
+		http.Error(writer, "No Organization ID Found", http.StatusBadRequest)
+		return
+	}
+	token := request.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)
+	userId := token.RegisteredClaims.Subject
+	readOrgPerm := fmt.Sprintf("org%s:read", organizationId)
+	canReadOrg, err := auth.HasPermission(userId, readOrgPerm)
+	if err != nil {
+		http.Error(writer, fmt.Sprintf("Failed to get user permissions: %s", err.Error()), http.StatusInternalServerError)
+		return
+	}
+	if !canReadOrg {
+		http.Error(writer, fmt.Sprintf("User does not have permission to read organization with id: %s", organizationId), http.StatusForbidden)
+		return
+	}
+
+	ctx := request.Context()
+	boardsInOrg, err := handler.controller.GetBoardsInOrg(ctx, organizationId)
+	if err != nil {
+		if err.Error() == sql.ErrNoRows.Error() {
+			http.Error(writer, fmt.Sprintf("No organization found with id %s", organizationId), http.StatusNotFound)
+		} else {
+			http.Error(writer, fmt.Sprintf("Failed to get organization: %s", err.Error()), http.StatusInternalServerError)
+		}
+		return
+	}
+	returnedBoards := []string{}
+	for _, element := range *boardsInOrg {
+		// TODO: CHECK IF THE BOARD IS PRIVATE
+		isBoardPrivate, err := handler.controller.CheckBoardPrivacy(ctx, element)
```

Comment:
This seems like excessive db querying. You are querying for a single column on the db when you get all the boards in the `GetBoardsInOrg` function. I feel that `GetBoardsInOrg` should have this for loop logic in it so you can minimize the querying. I guess this would make the `GetBoardsInOrg` function actually `GetViewableBoardsInOrg` but I think moving this logic onto the controller helps separate the code. I've been trying to keep all logic on the controller and the router is just data validation/auth validation then calling data through controller